src/hyperion/external_interaction/ispyb/store_datacollection_in_ispyb.py

```diff
@@ -173,22 +178,34 @@ def _store_data_collection_group_table(self, conn: Connector) -> int:
         params["sampleid"] = self.ispyb_params.sample_id
         params["sample_barcode"] = self.ispyb_params.sample_barcode
 
-        return mx_acquisition.upsert_data_collection_group(list(params.values()))
+        return self._upsert_data_collection_group(conn, params)
+
+    @staticmethod
+    def _upsert_data_collection_group(
+        conn: Connector, params: StrictOrderedDict
+    ) -> int:
+        return conn.mx_acquisition.upsert_data_collection_group(list(params.values()))
+
+    @staticmethod
+    def _upsert_data_collection(conn: Connector, params: StrictOrderedDict) -> int:
+        return conn.mx_acquisition.upsert_data_collection(list(params.values()))
 
     @TRACER.start_as_current_span("store_ispyb_data_collection_table")
     def _store_data_collection_table(
-        self, conn: Connector, data_collection_group_id: int
+        self,
+        conn: Connector,
+        data_collection_group_id: int,
+        data_collection_id: Optional[int] = None,
     ) -> int:
-        assert (
-            self.ispyb_params is not None
-            and self.detector_params is not None
-            and self.xtal_snapshots is not None
```

Comment:
Why do we no longer need to do the assert  on `self.xtal_snapshots`?

Reply:
I'm assuming that we won't have crystal snapshots at `begin_deposition` but perhaps I'm wrong? Are these fields supplied by GDA or is this something hyperion does?

Reply:
Not sure actually, I think GDA takes snapshots of a sample once it gets loaded, but I'm not sure which snapshots these are referring to. @dperl-dls will know!

Reply:
Hyperion takes the snapshots. As we currently stand we will have them at the start of deposition but I think it's reasonable to assume we might not always